### PR TITLE
ruby: Replace default tasks with a stub message

### LINKED ISSUE #12579: Ruby extension incorrectly detects tests
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The queries used to detect Minitest tests, RSpec example groups and RSpec specs are not sufficient to be included by default in the Ruby extension. Simply calling the method `test`, `it` or `describe` does not guarantee you are using Minitest or RSpec. Additionally, the definition of a method starting with `test_` does not guarantee the use of Minitest.

There are other test libraries for Ruby including [Sus](https://github.com/ioquatix/sus), [TLDR](https://github.com/tendersearls/tldr), and [Quickdraw](https://github.com/joeldrapper/quickdraw) that conflict with these queries.

Additionally, the commands that the runnables tasks trigger are not even correct for RSpec or Minitest. For RSpec, the command should start with `bundle exec rspec` to ensure the bundled version of RSpec is used. For minitest, this is more complicated. It’s usually `bundle exec rake` or `bundle exec rake test`. Sometimes it’s `bundle exec rails test` and sometimes (when working on a Rails gem), it’s `bin/test`.

## Steps to Reproduce
In any Ruby file, these three examples trigger runnables without correct scoping:

```ruby
def test_something
end

describe Something

test "something"
```

## Recommended solution

At minimum, the runnables should be removed by default. Perhaps they could be enabled with a configuration or by installing an `rspec` or a `minitest` extension. Alternatively, it might be possible to write more precise queries to detect the presence of the relevant Ruby gems.

I don’t know if this is possible, but the *ideal* solution would probably be to allow project-level configuration of tasks with the relevant runnable detection queries. That way, users could copy/paste e.g. the Minitest configuration into their project and update the task to execute `bundle exec rake` or `bin/test` or whatever they need.

### Environment

Zed: v0.137.6 (Zed)
OS: macOS 14.4.1
Memory: 128 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by vitallium**
The Ruby world has many testing frameworks:

- Minitest
- RSpec
- quickdraw
- tldr
- and many others.

Attempting to support all of them through a single `tasks.json` file is a challenging task and nearly impossible. All testing frameworks have different running options and commands. It's still possible to use tree-sitter queries to detect runnables in Ruby code but Zed lacks the ability to detect the testing framework in a project that can be used to detect the correct commands to run tests or runnables. The end user knows the correct command and it's wise to delegate creating the command to them. It would be a bit strange to leave the user without any guidance, so this commit adds example tasks for various Ruby testing frameworks.

Closes #12579

Here is the screenshot how it looks:

CleanShot 2024-07-01 at 19 37 08@2x

Release Notes:

- N/A

### CODE REVIEW

**Comment by vitallium** [comment]
"I will mark this PR as a draft while I wait for a review from @joeldrapper.

**Review by joeldrapper — APPROVED** [review]
Looks good to me.

**Comment by vitallium** [comment]
> Looks good to me.

Thank you! This PR is ready then.

**Comment by mikayla-maki** [comment]
Is there some collision in test patterns that make ruby test frameworks difficult to use? Because this looks like it's just removing information from the kinds of ruby test we can detect. Why not just add the tasks for each test framework?

**Review by mikayla-maki — CHANGES_REQUESTED** [review]
Is there some collision in test patterns that make ruby test frameworks difficult to use? Because this looks like it's just removing information from the kinds of ruby test we can detect. Why not just add the tasks for each test framework?

**Comment by joeldrapper** [comment]
> Why not just add the tasks for each test framework?

Mainly because there are so many different ways you might run tests in Ruby. For example, if you use Minitest, depending on how your project is setup, you might need to run any of the following commands to run your tests.

```
bundle exec rake
bundle exec rake test
bundle exec rails test
bin/rails test
bin/test
bundle exec rake parallel:test
```

I think the goal of this change is to surface information on how you can configure your own local test running task which the Ruby extension for Zed can use.

**Comment by mikayla-maki** [comment]
Ahhh, I see the problem now. Thanks for the explanation :)

**Pull request merged**
